### PR TITLE
feat: update workflow step action types [HOMER-612]

### PR DESCRIPTION
**Pull request by giotiskl**
Update workflow step action types so that users can create actions via the sdk

### CODE REVIEW

File: lib/entities/workflow-definition.ts [hunk 1]
```diff
@@ -14,14 +14,32 @@ import { wrapCollection } from '../common-utils'
...
-  action: Link<'AppAction'>
+export type WorkflowStepEmailActionConfiguration = {
+  recipients: EmailActionRecipient[]
+}
+
+export type WorkflowStepTaskActionConfiguration = {
+  assignee: Link<'User'> | Link<'Team'>
   body: string
-  headers: Record<string, unknown>
+  dueDate: number
```
Comment (ronaldronson) [inline lib/entities/workflow-definition.ts:39]: It's optional param.

**Review by ronaldronson — COMMENTED** [review]
Looks good, one small note.

File: lib/entities/workflow-definition.ts [hunk 2]
```diff
@@ -14,14 +14,32 @@ import { wrapCollection } from '../common-utils'
...
+
+type EmailActionRecipient = string | Link<'User'> | Link<'Team'>
 
-export type WorkflowStepAction = {
-  action: Link<'AppAction'>
+export type WorkflowStepEmailActionConfiguration = {
+  recipients: EmailActionRecipient[]
+}
+
+export type WorkflowStepTaskActionConfiguration = {
```
Comment (ronaldronson) [inline lib/entities/workflow-definition.ts:36]: Let's also add `WorkflowStepAppActionConfiguration` with an optional JSON body and headers. Basically, type the login presented here - https://github.com/contentful/workflow-management-api/blob/f2b7351152813b72da82b1548090e7a6d2ef9844/src/domain/workflow-definition/validations.ts#L217

Comment (giotiskl) [reply lib/entities/workflow-definition.ts:36]: done

File: lib/entities/workflow-definition.ts (hunk 1 above)
Comment (giotiskl) [reply lib/entities/workflow-definition.ts:39]: done

File: lib/entities/workflow-definition.ts [hunk 3]
```diff
@@ -14,14 +14,37 @@ import { wrapCollection } from '../common-utils'
...
+  assignee: Link<'User'> | Link<'Team'>
   body: string
-  headers: Record<string, unknown>
+  dueDate?: number
+}
+
+export type WorkflowStepAppActionConfiguration = {
+  body: Record<string, any>
```
Comment (ronaldronson) [inline lib/entities/workflow-definition.ts:43]: Both body and headers are optional params.

**Review by ronaldronson — CHANGES_REQUESTED** [review]

File: lib/entities/workflow-definition.ts (hunk 3 above)
Comment (giotiskl) [reply lib/entities/workflow-definition.ts]: done

File: lib/entities/workflow-definition.ts [hunk 4]
```diff
@@ -14,14 +14,37 @@ import { wrapCollection } from '../common-utils'
...
 import { ActionType } from './role'
 
-/* Workflow Step */
+/* Workflow Step Action */
+export enum WorkflowStepActionType {
+  App = 'app',
+  Email = 'email',
+  Task = 'task',
+}
+export interface WorkflowStepAction<T = Record<string, any>> {
```
Comment (Chaoste) [inline lib/entities/workflow-definition.ts:23]: Proposal for more explicit typings:
```
type WorkflowStepAction = WorkflowStepAppAction | WorkflowStepEmailAction | WorkflowStepTaskAction
type WorkflowStepAppAction = {
  appId: string
  appActionId: string
  type: 'app'
  configuration: { ... }
}
type WorkflowStepEmailAction = ...
type WorkflowStepTaskAction = ...
```

Comment (giotiskl) [reply lib/entities/workflow-definition.ts]: done

**Review by Chaoste — CHANGES_REQUESTED** [review]
Luv it :)

**Review by Chaoste — APPROVED** [review]
Upps

File: lib/entities/workflow-definition.ts [hunk 5]
```diff
@@ -14,14 +14,46 @@ import { wrapCollection } from '../common-utils'
...
+  }
+}
+
+type WorkflowStepAppAction = {
+  type: 'app'
+  appId: string
+  appActionId: string
+  configuration?: {
+    body?: Record<string, any>
+    headers?: Record<string, any>
```
Comment (ronaldronson) [inline lib/entities/workflow-definition.ts:52]: The value of the header is also a string, not any.

**Review by ronaldronson — CHANGES_REQUESTED** [review]
One last thing

File: lib/entities/workflow-definition.ts (hunk 5 above)
Comment (giotiskl) [reply lib/entities/workflow-definition.ts]: done

**Review by ronaldronson — APPROVED** [review]
That was fast

**Pull request merged**